packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -1,37 +1,170 @@
+import contextlib
+import json
 import logging
+import random
+import threading
+from collections import deque
 from subprocess import check_output
 
 import pytest
-
 import requests
-
 import retrying
 
 from pkgpanda.build import load_json
 from test_util.marathon import get_test_app, get_test_app_in_docker
 
 
+log = logging.getLogger(__name__)
+timeout = 500
+maxthreads = 8
+
+
 def lb_enabled():
     config = load_json('/opt/mesosphere/etc/expanded.config.json')
     return config['enable_lb'] == 'true'
 
 
-def ensure_routable(cmd, service_points, timeout=120):
-    @retrying.retry(wait_fixed=2000,
-                    stop_max_delay=timeout * 1000,
-                    retry_on_result=lambda ret: ret is False,
-                    retry_on_exception=lambda x: True)
-    def _ensure_routable():
-        proxy_uri = 'http://{}:{}/run_cmd'.format(service_points[0].host, service_points[0].port)
-        logging.info('Sending {} data: {}'.format(proxy_uri, cmd))
-        r = requests.post(proxy_uri, data=cmd)
-        logging.info('Requests Response: %s', repr(r.json()))
-        assert r.json()['status'] == 0
-    return _ensure_routable
+@retrying.retry(wait_fixed=2000,
+                stop_max_delay=timeout * 1000,
+                retry_on_result=lambda ret: ret is False,
+                retry_on_exception=lambda x: True)
+def ensure_routable(cmd, service_points):
+    proxy_uri = 'http://{}:{}/run_cmd'.format(service_points[0].host, service_points[0].port)
+    log.info('Sending {} data: {}'.format(proxy_uri, cmd))
+    r = requests.post(proxy_uri, data=cmd)
+    log.info('Requests Response: %s', repr(r.json()))
+    assert r.json()['status'] == 0
+    return json.loads(r.json()['output'])
+
+
+class VipTest:
+    def __init__(self, num, vip, vipaddr, samehost, vipnet, proxynet):
+        self.vip = vip.format(num, 7000 + num)
+        self.vipaddr = vipaddr.format(num, 7000 + num)
+        self.samehost = samehost
+        self.vipnet = vipnet
+        self.proxynet = proxynet
+
+    def __str__(self):
+        return ("VipTest(vip={},vipaddr={},samehost={},"
+                "vipnet={},proxynet={})").format(self.vip, self.vipaddr, self.samehost, self.vipnet, self.proxynet)
+
+    def log(self, s, lvl=logging.DEBUG):
+        m = 'VIP_TEST {} {}'.format(s, self)
+        log.log(lvl, m)
+
+
+def vip_test(dcos_api_session, r):
```

Comment:
Should have pointed this out earlier, this test is very `DockerContainerizer` specific. Would be a pain to change it if you want to replicate it for UCR. Can we modify it so that the same vip_test can be invoked with a paramter to test `DockerContainerizer` vs UCR?